src/Microsoft.VisualStudio.Composition/Reflection/PropertyRef.cs

```diff
@@ -32,25 +32,31 @@ public class PropertyRef : MemberRef, IEquatable<PropertyRef>
         /// </summary>
         private readonly int? setMethodMetadataToken;
 
-        public PropertyRef(TypeRef declaringType, int metadataToken, int? getMethodMetadataToken, int? setMethodMetadataToken, string name)
-            : base(declaringType, metadataToken)
+        public PropertyRef(TypeRef declaringType, TypeRef propertyTypeRef, int metadataToken, int? getMethodMetadataToken, int? setMethodMetadataToken, string name, bool isStatic)
+            : base(declaringType, metadataToken, isStatic)
         {
             this.getMethodMetadataToken = getMethodMetadataToken;
             this.setMethodMetadataToken = setMethodMetadataToken;
             this.Name = name;
+            this.PropertyTypeRef = propertyTypeRef;
         }
 
         public PropertyRef(PropertyInfo propertyInfo, Resolver resolver)
             : base(propertyInfo, resolver)
         {
+            this.getMethodMetadataToken = propertyInfo?.GetMethod?.MetadataToken;
+            this.setMethodMetadataToken = propertyInfo?.SetMethod?.MetadataToken;
             this.Name = propertyInfo.Name;
```

Comment:
I'm curious whether this was just an elective change or if there is some significance behind it that I can't see.

Reply:
It stops assembly loading in scenarios where either the get method or set method metadata is null. Right now if the metadata is intentionally null we will load the assembly this property is defined it just to make sure.